qiskit_experiments/analysis/fit_functions.py

```diff
@@ -0,0 +1,56 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A library of fit functions.
+"""
+# pylint: disable=invalid-name
+
+import numpy as np
+
+
+def cos(x: np.ndarray, amp: float, freq: float, phase: float, baseline: float) -> np.ndarray:
```

Comment:
Should some of these be kwargs with default values? Otherwise its difficult to use them for fit functions that don't use all the parameters: Eg something like
```suggestion
def cos(x: np.ndarray, amp: float = 1, freq: float = 1 / (2 * np.pi), phase: float = 0, baseline: float = 0) -> np.ndarray:
```

The `curve_fit` and `multi_curve_fit` functions also already work with kwarg functions and dicts for p0 and bounds so if you use the same params as these functions it should be easier. Though you might not want to use the parameter names of these library functions I guess.

Reply:
Make sense. I'll add default values and use the dict instead of ndarray.